.github/workflows/unit_tests.yaml

```diff
@@ -43,12 +43,16 @@ jobs:
         run: |
           echo "gemspec path: '../hammer-cli', name: 'hammer_cli'" > Gemfile.local.rb
           echo "gemspec path: '../hammer-cli-foreman', name: 'hammer_cli_foreman'" >> Gemfile.local.rb
+      - name: Remove Gemfile.lock
+        run: rm Gemfile.lock
       - name: Set up Ruby
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
           working-directory: hammer-cli-foreman-statistics
+      - name: Bundle Install
+        run: bundle install
```

Comment:
`ruby/setup-ruby` in the step above already does a `bundle install`, so you don't have to do it again (but it's also not the reason for the weird rake behaviour you asked about)